pkg/kv/kvserver/kvflowcontrol/rac2/priority.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package rac2
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/raft/raftpb"
+	"github.com/cockroachdb/cockroach/pkg/util/admission/admissionpb"
+)
+
+// AdmissionPriorityToRaftPriority maps the larger set of values in
+// admissionpb.WorkPriority to the smaller set of raftpb.Priority.
+func AdmissionPriorityToRaftPriority(pri admissionpb.WorkPriority) raftpb.Priority {
+	if pri < admissionpb.NormalPri {
+		return raftpb.LowPri
+	} else if pri < admissionpb.LockingNormalPri {
+		return raftpb.NormalPri
+	} else if pri < admissionpb.UserHighPri {
+		return raftpb.AboveNormalPri
+	} else if pri <= admissionpb.HighPri {
+		return raftpb.HighPri
+	} else {
```

Comment:
Do you know the most frequent one(s) here, for fast-path? Could place it first, and/or do something binary-search-like to reduce the max number of comparisons:

```golang
if pri < LockingNormalPri {
	if pri < NormalPri {
		return LowPri
	}
	return NormalPri
}
if pri < UserHighPri {
	return AboveNormalPri
}
return HighPri
```

Same below.